### PR TITLE
Fix battery temperature reporting invalid values in extreme temperatures

### PR DESCRIPTION
**Pull request by ghost**
### Description
Moving average logic has been fixed
-

### Requirements

- [x] Code style guidelines followed
- [ ] Documentation updated

### Tests 🧪

- [ ] Robot
- [ ] Container
- [ ] Simulation

### CODE REVIEW

File: husarion_ugv_utils/test/test_moving_average.cpp [hunk 1]
```diff
@@ -61,12 +61,12 @@ TEST(TestMovingAverage, TestHighOverload)
...
   double sum = 0.0;
   for (std::size_t i = 1; i <= window_len * 10; i++) {
-    sum += double(i);
+    sum += double(i) / double(window_len);
     ma.Roll(double(i));
 
     // test every 1000 rolls expected average
     if (i % window_len == 0) {
-      EXPECT_EQ(sum / double(window_len), ma.GetAverage());
+      EXPECT_LT(sum - ma.GetAverage(), std::numeric_limits<double>::epsilon());
```
Comment (rafal-gorecki) [inline husarion_ugv_utils/test/test_moving_average.cpp]: Please use `ASSERT_NEAR(val1, val2, abs_error);` function if you'd like to compare two values with some treshold.

File: husarion_ugv_utils/include/husarion_ugv_utils/moving_average.hpp [hunk 2]
```diff
@@ -25,40 +25,40 @@ class MovingAverage
 {
 public:
   MovingAverage(const std::size_t window_size = 5, const T initial_value = T(0))
-  : window_size_(window_size), initial_value_(initial_value), sum_(T(0))
+  : window_size_(window_size), initial_value_(initial_value)
   {
   }
```
Comment (rafal-gorecki) [inline husarion_ugv_utils/include/husarion_ugv_utils/moving_average.hpp]:
```suggestion
  {
    if (window_size_ == 0) {
    throw std::invalid_argument("Window size must be greater than 0");
    }
  }
```

File: husarion_ugv_utils/include/husarion_ugv_utils/moving_average.hpp [hunk 3]
```diff
@@ -25,40 +25,40 @@ class MovingAverage
...
+    for (const auto & value : values_) {
+      sum += value / static_cast<T>(values_.size());
+    }
+
+    return sum;
   }
 
 private:
   const std::size_t window_size_;
   std::deque<T> values_;
```
Comment (rafal-gorecki) [inline husarion_ugv_utils/include/husarion_ugv_utils/moving_average.hpp]: Optional. IMO `buffer_` is more better name, than `values_`.

File: husarion_ugv_utils/include/husarion_ugv_utils/moving_average.hpp [hunk 4]
```diff
@@ -25,40 +25,40 @@ class MovingAverage
...
     if (values_.size() == 0) {
       return initial_value_;
     }
-    return sum_ / static_cast<T>(values_.size());
+
+    T sum = T(0);
+
+    for (const auto & value : values_) {
+      sum += value / static_cast<T>(values_.size());
+    }
```
Comment (rafal-gorecki) [inline husarion_ugv_utils/include/husarion_ugv_utils/moving_average.hpp:53]:
```suggestion
    T sum = std::accumulate(buffer_.begin(), buffer_.end(), initial_value_);
```

File: husarion_ugv_utils/include/husarion_ugv_utils/moving_average.hpp [hunk 5]
```diff
@@ -25,40 +25,40 @@ class MovingAverage
...
       values_.pop_front();
     }
   }
 
-  void Reset()
-  {
-    values_.erase(values_.begin(), values_.end());
-    sum_ = T(0);
-  }
+  void Reset() { values_.erase(values_.begin(), values_.end()); }
```
Comment (rafal-gorecki) [inline husarion_ugv_utils/include/husarion_ugv_utils/moving_average.hpp:41]: Reset probably also should be able to set `initial_value_`

File: husarion_ugv_utils/include/husarion_ugv_utils/moving_average.hpp [hunk 6]
```diff
@@ -25,40 +25,40 @@ class MovingAverage
...
     }
-    return sum_ / static_cast<T>(values_.size());
+
+    T sum = T(0);
+
+    for (const auto & value : values_) {
+      sum += value / static_cast<T>(values_.size());
+    }
+
+    return sum;
```
Comment (rafal-gorecki) [inline husarion_ugv_utils/include/husarion_ugv_utils/moving_average.hpp]: Devision is computation expensive we can first acumulate buffor and then devide. Reduce devision from N -> 1.
```suggestion
    T sum = std::accumulate(buffer_.begin(), buffer_.end(), 0);
    T average = sum / static_cast<T>(values_.size());

    return average;
```

**Review by rafal-gorecki — CHANGES_REQUESTED** [review]
Some suggestions

File: husarion_ugv_utils/include/husarion_ugv_utils/moving_average.hpp (hunk 6 above)
Comment (ghost) [reply husarion_ugv_utils/include/husarion_ugv_utils/moving_average.hpp]: I did it this way to improve numeric stability for floats with large accumulate buffer values, but honestly without using recursive average or more numerically stable summation methods it does not matter that much.

**Review by rafal-gorecki — APPROVED** [review]
Merge after test.

**Comment by ghost** [comment]
Works on real hardware, merging

**Pull request merged**